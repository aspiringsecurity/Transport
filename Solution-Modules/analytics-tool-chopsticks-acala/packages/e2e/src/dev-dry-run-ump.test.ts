import { describe, expect, it } from 'vitest'

import { setupApi, ws } from './helper'

setupApi({
  endpoint: 'wss://rpc.polkadot.io',
  blockHash: '0xb012d04c56b65cfa1f47cb1f884d920f95d0097b1ed42f5da18d5e2a436c2f3e',
})

describe('dev_dryRun ump', () => {
  it('works', async () => {
    const params = [
      {
        raw: false,
        ump: {
          // https://acala.subscan.io/xcm_message/polkadot-ff66f28818d0b74573e62db8317e354b253fbc80
          2000: [
            '0x021000040000000007903fc4db080a130000000007903fc4db08000d010004000101009c4b11a0974cba4a395c94832fba812868a6cb0ba09e8519b3521093ea359905',
          ],
        },
      },
    ]
    const resp = await ws.send('dev_dryRun', params)
    expect(resp.new.system.events).toMatchSnapshot()
  })
})
