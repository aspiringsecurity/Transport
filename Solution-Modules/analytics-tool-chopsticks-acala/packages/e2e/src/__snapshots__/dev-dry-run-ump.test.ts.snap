// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dev_dryRun ump > works 1`] = `
[
  {
    "event": {
      "data": {
        "dispatchInfo": {
          "class": "Mandatory",
          "paysFee": "Yes",
          "weight": {
            "proofSize": "0",
            "refTime": "233,042,000",
          },
        },
      },
      "index": "0x0000",
      "method": "ExtrinsicSuccess",
      "section": "system",
    },
    "phase": {
      "ApplyExtrinsic": "0",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "38,046,810,000",
        "who": "13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy",
      },
      "index": "0x0508",
      "method": "Withdraw",
      "section": "balances",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "37,638,163,032",
        "who": "14Xvmfh7QYWYJUUkLHVmyRkJ1PbLLNHJ1RnsnsZ1Ph7pNxg3",
      },
      "index": "0x0507",
      "method": "Deposit",
      "section": "balances",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "amount": "408,646,968",
        "who": "1XQn94kWaMVJG16AWPKGmYFERfttsjZq4ompSTz2jxHK6uL",
      },
      "index": "0x0507",
      "method": "Deposit",
      "section": "balances",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": [
        "0x48079e14ffeb3fb6f43147677425488c414c9f58d90b608de5f34f0ac8df8d50",
        {
          "Complete": "4,000,000,000",
        },
      ],
      "index": "0x3b02",
      "method": "ExecutedUpward",
      "section": "ump",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
  {
    "event": {
      "data": {
        "dispatchInfo": {
          "class": "Mandatory",
          "paysFee": "Yes",
          "weight": {
            "proofSize": "0",
            "refTime": "97,884,000",
          },
        },
      },
      "index": "0x0000",
      "method": "ExtrinsicSuccess",
      "section": "system",
    },
    "phase": {
      "ApplyExtrinsic": "1",
    },
    "topics": [],
  },
]
`;
